includes/HttpFunctions.php

```diff
@@ -296,6 +296,16 @@ function __construct( $url, $options = array() ) {
 			$this->status = Status::newGood( 100 ); // continue
 		}
 
+		// SUS-5499: Use internal host name for MW->MW requests when running on Kubernetes
+		if ( $wgKubernetesNamespace && Http::isLocalURL( $url ) ) {
+			$targetHost = $this->parsedUrl['host'];
+
+			$this->url = str_replace( $targetHost, "$wgKubernetesDeploymentName.$wgKubernetesNamespace", $this->url );
+			$this->parsedUrl['host'] = "$wgKubernetesDeploymentName.$wgKubernetesNamespace";
+
+			$this->setHeader( 'Host', $targetHost );
+		}
+
```

Comment:
Maybe let's use proxy instead?

Reply:
hmm, but what would we be proxying through? borders will cause the request to go to apaches.